api/src/main/java/com/epam/pipeline/manager/cluster/NodesManager.java

```diff
@@ -181,7 +182,13 @@ public NodeInstance getNode(String name, FilterPodsRequest request) {
                 );
                 return Optional.of(nodeInstance);
             }
+        } catch (KubernetesClientException e) {
+            return missingNodeInstance(name, request);
```

Comment:
Maybe just log exception and use `return` outside of catch clause.

Reply:
Done.